cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/SourceControlManagementService.java

```diff
@@ -141,7 +143,8 @@ public RepositoryMeta setRepository(NamespaceId namespace, RepositoryConfig repo
    * @param namespace {@link NamespaceId} do unlink repository from
    */
   public void deleteRepository(NamespaceId namespace) {
-    accessEnforcer.enforce(namespace, authenticationContext.getPrincipal(), StandardPermission.DELETE);
+    // Deleting on namespace children resources is translated to Update on namespace entity
+    accessEnforcer.enforce(namespace, authenticationContext.getPrincipal(), StandardPermission.UPDATE);
```

Comment:
@tivv fyi, updated to `StandardPermission.UPDATE`, but still have a question here: should we instead enforce `NamespacePermission.UPDATE_REPOSITORY_METADATA` here instead? Otherwise roles like developer cannot edit the config but can delete it...

Reply:
Updated to use `UPDATE_REPOSITORY_METADATA`